test/active_record_doctor/detectors/extraneous_indexes_test.rb

```diff
@@ -61,6 +61,46 @@ def test_multi_column_covered_by_unique_and_non_unique_multi_column_is_duplicate
 OUTPUT
   end
 
+  def test_not_covered_by_different_index_type
+    create_table(:users) do |t|
+      t.string :first_name
+      t.string :last_name
+      t.index [:last_name, :first_name], using: :btree
+      t.index :last_name, using: :hash
```

Comment:
While technically the second index is covered by the first, I think, originally `ExtraneousIndexes` check was meant to work with B-trees only, as this could cover 99% of the cases and makes its implementation simpler.

So I limited it's checks only for B-tree indexes and assume differing index types as incompatible.